core/templates/pages/story-editor-page/navbar/story-editor-navbar.component.spec.ts

```diff
@@ -719,55 +718,89 @@ describe('Story editor navbar component', () => {
     let publishStorySpy = spyOn(component, 'publishStory');
-    let saveChangesSpy = spyOn(component, 'saveChanges');
+    let loadStorySpy = spyOn(
+      storyEditorStateService, 'loadStory').and.returnValue();
+    let alertServiceSpy = spyOn(
+      alertsService, 'addInfoMessage').and.callThrough();
+    let check: number = 0;
+    let saveStorySpy = spyOn(storyEditorStateService, 'saveStory')
+      .and.callFake((commitMessage, successCallback, errorCallback) => {
+        if (check) {
+          successCallback();
+        } else {
+          errorCallback('Error');
+        }
+        storyEditorStateService.setChapterStatusIsChanging(false);
+        return true;
+      });
 
     component.changeChapterStatus('Published');
     expect(storyNodeStatusSpy).toHaveBeenCalledTimes(1);
     expect(storyNodeUnpublishingReasonSpy).toHaveBeenCalledTimes(1);
-    expect(saveChangesSpy).toHaveBeenCalled();
+    expect(chapterStatusChangingSpy).toHaveBeenCalledTimes(2);
+    expect(alertServiceSpy).toHaveBeenCalled();
+    expect(saveStorySpy).toHaveBeenCalled();
 
 
     component.story.getStoryContents().getNodes()[0].
       setStatus('Ready To Publish');
     component.story.getStoryContents().getNodes()[0].
       setFirstPublicationDateMsecs(null);
+    check = 1;
 
     component.changeChapterStatus('Published');
     expect(storyNodeStatusSpy).toHaveBeenCalledTimes(3);
     expect(storyNodeUnpublishingReasonSpy).toHaveBeenCalledTimes(3);
     expect(storyNodeFirstPublicationDateSpy).toHaveBeenCalledTimes(1);
-    expect(saveChangesSpy).toHaveBeenCalled();
+    expect(chapterStatusChangingSpy).toHaveBeenCalledTimes(4);
+    expect(saveStorySpy).toHaveBeenCalled();
     expect(publishStorySpy).toHaveBeenCalled();
+    expect(loadStorySpy).toHaveBeenCalled();
   });
 
   it('should unpublish chapters', fakeAsync(() => {
-    component.story = Story.createFromBackendDict(storyBackendDict);
+    story = Story.createFromBackendDict(storyBackendDict);
+    spyOn(storyEditorStateService, 'getStory').and.returnValue(story);
+    component.story = story;
     spyOn(
       storyEditorStateService,
       'getSelectedChapterIndexInPublishUptoDropdown').and.returnValue(0);
     let storyNodePlannedPublicationDateSpy = spyOn(
       storyUpdateService, 'setStoryNodePlannedPublicationDateMsecs');
+    let chapterStatusChangingSpy = spyOn(
+      storyEditorStateService, 'setChapterStatusIsChanging');
     let storyNodeStatusSpy = spyOn(storyUpdateService, 'setStoryNodeStatus');
     let storyNodeUnpublishingReasonSpy = spyOn(
       storyUpdateService, 'setStoryNodeUnpublishingReason');
-    const saveChangesSpy = spyOn(
+    let unpublishStorySpy = spyOn(component, 'unpublishStory');
+    let loadStorySpy = spyOn(
+      storyEditorStateService, 'loadStory').and.returnValue();
+    let alertServiceSpy = spyOn(
+      alertsService, 'addInfoMessage').and.callThrough();
+    let check: number = 0;
```

Comment:
This isn't named clearly enough, please use a clearer name.

Reply:
Done